### PR TITLE
New Entry and Terms General: Programming Paradigms 

### PR DESCRIPTION
**Pull request by yangc95**
### Description

- Created a new entry in General on programming paradigms.
- Nested functional programming and object oriented programming entries into the terms folder for programming paradigms.

### Type of Change

- Adding a new entry
- Adding a term
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/programming-paradigms/programming-paradigms.md
Comment (SSwiniarski) [inline content/general/concepts/programming-paradigms/programming-paradigms.md:10]: We already have an OOP tag. Adding the tag spelled out is going to affect a lot of entries that already have the OOP tag.

File: documentation/tags.md
Comment (SSwiniarski) [inline documentation/tags.md:189]: Maybe just stick with the OOP tag?

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@yangc95, I did a first review and left a couple of comments.

File: content/general/concepts/programming-paradigms/programming-paradigms.md
Comment (yangc95) [inline content/general/concepts/programming-paradigms/programming-paradigms.md:10]:
```suggestion
  - 'OOP'
```

File: documentation/tags.md
Comment (yangc95) [inline documentation/tags.md:189]:
```suggestion
```

File: content/general/concepts/programming-paradigms/programming-paradigms.md
Comment (yangc95) [inline content/general/concepts/programming-paradigms/programming-paradigms.md:18]:
```suggestion
Moving from low-level languages (e.g. [machine code](https://www.codecademy.com/resources/docs/general/machine-code)) to high-level languages (e.g. [Python](https://www.codecademy.com/resources/docs/python)), programming paradigms change.
```

**Comment by yangc95** [comment]
@SSwiniarski Thanks for conducting the first review! I incorporated your comments and suggestions. Please LMK if anything else needs adjusting

**Review by SSwiniarski — APPROVED** [review]
@yangc95 Looks good from here! Ready for a second review.

File: content/general/concepts/programming-paradigms/programming-paradigms.md
Comment (Dusch4593) [inline content/general/concepts/programming-paradigms/programming-paradigms.md:18]:
```suggestion
Moving from low-level languages (e.g., [machine code](https://www.codecademy.com/resources/docs/general/machine-code)) to high-level languages (e.g., [Python](https://www.codecademy.com/resources/docs/python)), programming paradigms change.
```

File: content/general/concepts/programming-paradigms/programming-paradigms.md
Comment (Dusch4593) [inline content/general/concepts/programming-paradigms/programming-paradigms.md:16]:
```suggestion
A **programming paradigm** is an approach to programming. Depending on the language, the preferred paradigm may differ. Some languages, such as [C++](https://www.codecademy.com/resources/docs/cpp) and [JavaScript](https://www.codecademy.com/resources/docs/javascript), cover multiple paradigms.
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@yangc95 Left some comments/suggestions below. Could we also add the `'Functional Programming'` tag to `fuctional-programming.md` and `'OOP'` to `object-oriented-programming.md`?

**Comment by yangc95** [comment]
@Dusch4593 @SSwiniarski does everything look good here before we merge?

**Comment by SSwiniarski** [comment]
My only reservation is do we have any inbound links to either object oriented programming or functional programming that this breaks? If so, we need to add those links to this PR.

**Review by Dusch4593 — APPROVED** [review]
@yangc95 @SSwiniarski Everything LGTM! I went in and fixed the links in other entries that would have stopped working, otherwise. 😄

**Pull request merged**
